Feature/graphql vendors refactor

lafbelmonte opened the pull request:
-Updated use case return to boolean for graphql mutations
-Updated all tests to adopt new use case
-Used chance to mock name
-Change type of id to ID! in type definitions

lafbelmonte committed:
feat(lib): updated return of use case

-updated to boolean for mutations
-updated id type to ID! in typeDefs

lafbelmonte committed:
test: updated all tests

-updated test for new boolean return use case
-used chance for mocking name

lafbelmonte committed:
feat(library): added mongoose-lean-virtuals

-added mongoose-lean-virtuals to use virtuals on lean

ralphcasipe1 commented on koa-crud/src/graphql/typeDefs.ts:
```diff
@@ -2,27 +2,17 @@ import { gql } from 'apollo-server-koa';
-  type VendorReturn {
-    message: String
-    data: Vendor
+    createVendor(name: String, type: String): Boolean
+    updateVendor(id: ID!, name: String, type: String): Boolean
```
please add input for `createVendor` and `updateVendor`

e.g.

```gql
input CreateVendorInput {
  name: String!
  type: VendorType!
}

type Mutation {
   createVendor(input: CreateVendorInput!): Boolean
}
```

lafbelmonte replied:
noted sir

ralphcasipe1 commented on koa-crud/src/data-access/mongoose/vendors/actions.ts:
```diff
@@ -12,11 +12,11 @@ const actions = ({ Vendor }): VendorStore => {
 
   async function selectOneVendorByFilters(filters) {
-    return Vendor.findOne({ ...filters }).lean();
+    return Vendor.findOne({ ...filters }).lean({virtuals: true});
```
If you are only using `filters` here
```ts
{ ...filters }
```

You can also apply it directly
```ts
    return Vendor.findOne(filters).lean({virtuals: true});

```

lafbelmonte replied:
noted sir

ralphcasipe1 commented on koa-crud/src/graphql/typeDefs.ts:
```diff
@@ -2,27 +2,17 @@ import { gql } from 'apollo-server-koa';
-    _id: ID!
+    id: ID!
     name: String
     type: String
```
Add a enum `VendorType`
```gql
enum VendorType {
  SEAMLESS
  TRASFER
}
```

lafbelmonte replied:
noted sir

ralphcasipe1 reviewed (changes requested):
Next time please, make sure to have only one intent in a feature. As much as possible name your feature branch base on an intent for example:
- `feature/adopt-graphql`
- `feature/update-test-use-case`

This will help reviewers focus on one thing and minimizes mental mapping

lafbelmonte committed:
feat(lib): added input type for mutations

- added input for create vendor and update vendor
-removed unnecessary spread syntax
